### PR TITLE
button pagepart

### PR DESCRIPTION
**Pull request by denbatte**
Button pagepart in default website

### CODE REVIEW

**Comment by denbatte** [comment]
@kimausloos

File: src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+     * @var boolean
+     *
+     * @ORM\Column(name="center", type="boolean", nullable=true)
+     */
+    private $center = false;
+
+    /**
+     * @var array Supported types
+     */
+    public static $TYPES = array('primary','secondary','tertiary','quaternary','quinary','link');
```
Comment (kimausloos) [inline src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php]: No need for caps

File: src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+
+    /**
+     * @var array Supported sizes
+     */
+    public static $SIZES = array('xl','lg','default','sm','xs');
```
Comment (kimausloos) [inline src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php]: No need for caps

File: src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php [hunk 3]
```diff
@@ -0,0 +1,297 @@
...
+    /**
+     * Set type
+     *
+     * @param string $type
+     * @return ButtonPagePart
+     */
+    public function setType($type)
+    {
+        if (!in_array($type, self::$TYPES)) {
+            $type = 'primary';
```
Comment (kimausloos) [inline src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php]: imho this should throw an exception(fast) or trigger a validation error (clean)

File: src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php [hunk 4]
```diff
@@ -0,0 +1,297 @@
...
+    /**
+     * Set size
+     *
+     * @param string $size
+     * @return ButtonPagePart
+     */
+    public function setSize($size)
+    {
+        if (!in_array($size, self::$SIZES)) {
+            $size = 'default';
```
Comment (kimausloos) [inline src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php]: imho this should throw an exception(fast) or trigger a validation error (clean)

File: src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php [hunk 5]
```diff
@@ -0,0 +1,297 @@
...
+     */
+    public function getDefaultAdminType()
+    {
+        return new ButtonPagePartAdminType();
+    }
+
+    /**
+     * @return string
+     */
+    public function getCssClasses()
```
Comment (kimausloos) [inline src/Kunstmaan/PagePartBundle/Entity/ButtonPagePart.php]: This is not something an entity should care about

**Pull request closed without merging**